client.go

```diff
@@ -519,6 +538,10 @@ func (client *Client) Capture(packet *Packet, captureTags map[string]string) (ev
 		return
 	}
 
+	if mrand.Float32() > client.sampleRate {
```

Comment:
Can we add a check for:

```go
if client.sampleRate < 1.0 && mrand.Float32() > client.sampleRate
```

This will skip even generating a random number, which is relatively slow, unless we are sampling.

Reply:
nice catch!